python/ml4ir/applications/ranking/tests/test_feature_layer.py

```diff
@@ -282,10 +282,81 @@ def test_categorical_embedding_with_vocabulary_file(self):
+        #####################################################
+        embedding_size = 32
+        dropout_rate = 0.999
+        feature_info = {
+            "name": "categorical_variable",
+            "feature_layer_info": {
+                "fn": "categorical_embedding_with_vocabulary_file",
+                "args": {
+                    "vocabulary_file": "ml4ir/applications/ranking/tests/data/config/domain_name_vocab.csv",
+                    "embedding_size": embedding_size,
+                    "dropout_rate": dropout_rate,
+                },
+            },
+        }
+
+        # Define an input string tensor
+        string_tensor = tf.constant(
+            ["domain_0", "domain_1", "domain_0", "domain_2", "domain_10", "domain_11"]
+        )
+        try:
+            categorical_embedding = categorical_fns.categorical_embedding_with_vocabulary_file_and_dropout(
+                string_tensor, feature_info, self.file_io
+            )
+        except ValueError:
+            # Should throw error as method does not work with IDs containing 0
+            assert True
+
+        ########################################################
+        # Test for vocabulary file with no ids mapping specified
+        ########################################################
+        feature_info["feature_layer_info"]["args"][
+            "vocabulary_file"
+        ] = "ml4ir/applications/ranking/tests/data/config/domain_name_vocab_no_id.csv"
+
+        categorcial_tensor = tf.keras.Input(shape=(1,), dtype=tf.string)
+        embedding_tensor = categorical_fns.categorical_embedding_with_vocabulary_file(
+            categorcial_tensor, feature_info, self.file_io
+        )
+        model = tf.keras.model(categorcial_tensor, embedding_tensor)
+
+        categorical_embedding = model(string_tensor, training=False)
+
+        # Assert the right shapes of the resulting embedding
+        assert categorical_embedding.shape[0] == len(string_tensor)
+        assert categorical_embedding.shape[1] == 1
+        assert categorical_embedding.shape[2] == embedding_size
+
+        # Strings 0 and 2 should result in the same embedding because they are the same
+        assert tf.reduce_all(tf.equal(categorical_embedding[0], categorical_embedding[2]))
+        assert not tf.reduce_all(tf.equal(categorical_embedding[0], categorical_embedding[1]))
+        assert tf.reduce_all(tf.equal(categorical_embedding[4], categorical_embedding[5]))
+
+        # Strings domain_0 and domain_2 should NOT result in the same embedding because they use a default one-to-one vocabulary mapping
+        assert not tf.reduce_all(tf.equal(categorical_embedding[0], categorical_embedding[3]))
+        assert not tf.reduce_all(tf.equal(categorical_embedding[3], categorical_embedding[4]))
+
+        categorical_embedding = model(string_tensor, training=True)
+
+        # Since dropout_rate is set to 0.999, all categorical indices
```

Comment:
@jakemannix I've set the `dropout_rate = 0.999` to a really high value to test that every org gets mapped to OOV token at training time.

Reply:
No. It just doesn't take a value of 1.0 unfortunately. :/